internal/compiler-bridge/src/main/scala/xsbt/Console.scala

```diff
@@ -0,0 +1,133 @@
+  //       }
+  //       interpreter.setContextClassLoader()
+  //     } else
+  //       super.createInterpreter()
+  //
+  //     def bind(values: Seq[(String, Any)]) {
+  //       // for 2.8 compatibility
+  //       final class Compat {
+  //         def bindValue(id: String, value: Any) =
+  //           interpreter.bind(id, value.asInstanceOf[AnyRef].getClass.getName, value)
+  //       }
+  //       implicit def compat(a: AnyRef): Compat = new Compat
+  //       for ((id, value) <- values)
+  //         interpreter.beQuietDuring(interpreter.bindValue(id, value))
+  //     }
+  //
+  //     bind(bindNames zip bindValues)
+  //
+  //     if (!initialCommands.isEmpty)
+  //       interpreter.interpret(initialCommands)
+  //   }
+  //   override def closeInterpreter() {
+  //     if (!cleanupCommands.isEmpty)
+  //       interpreter.interpret(cleanupCommands)
+  //     super.closeInterpreter()
+  //   }
+  // }
+
+  val interpreter: IMain = new IMain(compilerSettings, new PrintWriter(outWriter)) {
+    def lastReq = prevRequestList.last
+  }
+
+  // val interpreter = new Interpreter(compilerSettings) {
+  // TODO: Fix this
+  // override protected def parentClassLoader = if (loader eq null) super.parentClassLoader else loader
+  // override protected def newCompiler(settings: Settings, reporter: Reporter) = super.newCompiler(compilerSettings, reporter)
+  //}
+  override def interpret(line: String, synthetic: Boolean): ConsoleResponse =
+    {
+      clearBuffer()
+      val r = interpreter.interpret(line, synthetic)
+      ConsoleResponse(fromScalaInterpreterResult(r), outWriter.toString)
+    }
+  def clearBuffer(): Unit = {
+    // errorWriter.getBuffer.setLength(0)
+    outWriter.getBuffer.setLength(0)
+  }
+
+  def reset(): Unit = {
+    clearBuffer()
+    interpreter.reset()
+  }
+
+  def fromScalaInterpreterResult(result: scala.tools.nsc.interpreter.IR.Result) = result match {
+    case Success    => ConsoleResult.Success
+    case Error      => ConsoleResult.Error
+    case Incomplete => ConsoleResult.Incomplete
+  }
+}
+
+object MakeSettings {
+  def apply(args: List[String], onError: String => Unit) =
+    {
+      val command = new GenericRunnerCommand(args, onError(_))
+      if (command.ok) command.settings
+      // TODO: Provide better exception
+      else throw new Exception(command.usageMsg)
+    }
+
+  def sync(args: Array[String], bootClasspathString: String, classpathString: String, onError: String => Unit): Settings =
+    {
+      val compilerSettings = sync(args.toList, onError)
+      if (!bootClasspathString.isEmpty)
+        compilerSettings.bootclasspath.value = bootClasspathString
+      compilerSettings.classpath.value = classpathString
+      compilerSettings
+    }
+
+  def sync(options: List[String], onError: String => Unit) = {
+    val settings = apply(options, onError)
+
+    // -Yrepl-sync is only in 2.9.1+
```

Comment:
We can remove this tiny compat bit, we don't support Scala 2.9.x anymore :wink:.